smslibrary/src/main/java/com/eis/smslibrary/SMSPart.java

```diff
@@ -0,0 +1,60 @@
+package com.eis.smslibrary;
+
+import java.util.Comparator;
+
+/**
+ * Object associating to each part of a multi-part SMS text message:
+ * - the corresponding Intent's action to be received by SMSDeliveredBroadcastReceiver or
+ * SMSSentBroadcastReceiver;
+ * - whether the Intent was received;
+ *
+ * @author Giovanni Velludo
+ */
+class SMSPart implements Comparable<SMSPart>, Comparator<SMSPart> {
```

Comment:
why comparator?

Reply:
`Collections.binarySearch()` needs the object which is being searched for to implement `Comparator`. I use that method in `SMSSentBroadcastReceiver` and `SMSDeliveredBroadcastReceiver`.

Reply:
you give it 2 params
so you use 
public static <T> int binarySearch(List<T extends Comparable> list, T key)
and not 
public static <T> int binarySearch(List<T> list, T key, Comparator<T> c)

Reply:
Right, I'll remove `Comparator`